vm/qemu/qemu.go

```diff
@@ -601,11 +658,20 @@ func (inst *instance) Run(timeout time.Duration, stop <-chan bool, command strin
 }
 
 func (inst *instance) Diagnose() ([]byte, bool) {
-	select {
-	case inst.diagnose <- true:
-	default:
+	msg := fmt.Sprintf("%s Registers:\n", time.Now().Format("15:04:05 "))
+	ret := []byte{}
+	ret = append(ret, []byte(msg)...)
+	for cpu := 0; cpu < inst.cfg.CPU; cpu++ {
+		regs, err := inst.hmp("info registers", cpu)
+		if err == nil {
+			log.Logf(0, "VM-%v\n%v", inst.index, regs)
+			ret = append(ret, []byte(regs)...)
+		} else {
+			log.Logf(0, "VM-%v failed reading regs: %v",
+					inst.index, err)
+		}
 	}
-	return nil, false
+	return ret, true
```

Comment:
Why change from false to true?

Reply:
"true" makes vm/vm.go:MonitorExecution()  call mon.waitForOutput() which I thought is a good thing, no?

Reply:
We kill the qemu process, so there won't be any more output. In the next patch you also do the exit command, which I assume has similar effect.

Reply:
We do not kill it for 10 seconds  when returned "true" here so the VM kernel can still print stuff. Potentially we could inject "nmi" which QEMU injects into the VM as a machine check stop and this will produce more output. I should probably add "nmi" too but not quite sure what x86's analog would do the same.

Reply:
If we collect all diagnostics synchronously and return them from this function, returning true does not make sense.
Returning true only makes sense it we triggered some asynchronous debugging output in the VM and just return.

Reply:
Injecting nmi to a VM produces different results depending on the VM's kernel version, config, command line and system settings. Hard to tell if the kernel stopped printing in response to "nmi" or not. If it is just defaults, it prints register dump but if ftrace is enabled, it is going to take up to a minute but since the timeout is just 10sec, it is hardly useful so I'll drop this change for now.

Reply:
We also don't inject NMI as far as I see :)